libraries/AP_Beacon/AP_Beacon_Nooploop.h

```diff
@@ -0,0 +1,85 @@
+#pragma once
+
+#include "AP_Beacon_Backend.h"
+
+#define NOOPLOOP_MSG_BUF_MAX      256
+
+#define NOOPLOOP_HEADER                     0x55    // message header
+#define NOOPLOOP_FUNCTION_MARK_NODE_FRAME2  4
+#define NOOPLOOP_NODE_FRAME2_FRAMELEN_MAX   4096    // frames should be less than 4k bytes
+#define NOOPLOOP_NODE_FRAME2_SYSTIME        6       // start of 4 bytes holding system time in ms
+#define NOOPLOOP_NODE_FRAME2_PRECISION_X    10      // start of 1 byte holding precision in m*100 in x axis
+#define NOOPLOOP_NODE_FRAME2_PRECISION_Y    11      // start of 1 byte holding precision in m*100 in y axis
+#define NOOPLOOP_NODE_FRAME2_PRECISION_Z    12      // start of 1 byte holding precision in m*100 in y axis
+#define NOOPLOOP_NODE_FRAME2_POSX           13      // start of 3 bytes holding x position in m*1000
+#define NOOPLOOP_NODE_FRAME2_POSY           16      // start of 3 bytes holding y position in m*1000
+#define NOOPLOOP_NODE_FRAME2_POSZ           19      // start of 3 bytes holding z position in m*1000
+#define NOOPLOOP_NODE_FRAME2_VALID_NODES    118
+#define NOOPLOOP_NODE_FRAME2_NODE_BLOCK     119
+
+#define NOOPLOOP_HEADER2                    0x54    // message header
+#define NOOPLOOP_FUNCTION_MARK_SETTING_FRAME0  0
+#define NOOPLOOP_SETTING_FRAME0_A0          37
+
+class AP_Beacon_Nooploop : public AP_Beacon_Backend
+{
+
+public:
+    AP_Beacon_Nooploop(AP_Beacon &frontend, AP_SerialManager &serial_manager);
+
+    // return true if sensor is basically healthy (we are receiving data)
+    bool healthy() override;
+
+    // update the state of the sensor
+    void update() override;
+
+private:
+    enum class MsgType : uint8_t {
+        INVALID = 0,
+        NODE_FRAME2,
+        SETTING_FRAME0
+    };
+
+    // process one byte received on serial port
+    // message is stored in _msgbuf
+    MsgType parse_byte(uint8_t b);
+
+    // send setting_frame0 to tag. tag will ack setting_frame0 with anchor position filled
+    void request_setting();
+    
+    // pase node_frame2 to get tag position and distance
+    void parse_node_frame2();
+    
+    // parse setting_frame0 to get anchor position
+    void parse_setting_frame0();
+
+    // enum of valid function marks
+    enum class FunctionMark : uint8_t {
+        ANCHOR_FRAME0 = 0x00,
```

Comment:
As @peterbarker brought up in the dev call, it is quite unusual to have two items in an enum with the same value.  We think one of them isn't actually used and could be removed.

Reply:
I found this enum is not used anywhere. Maybe we can just remove it?

Reply:
yes, the suggestion that it could just be removed also came up on the call.